codegen-core/src/main/kotlin/software/amazon/smithy/rust/codegen/core/smithy/generators/EnumGenerator.kt

```diff
@@ -104,50 +118,21 @@ open class EnumGenerator(
     }
 
     open fun render() {
-        if (enumTrait.hasNames()) {
-            // pub enum Blah { V1, V2, .. }
-            renderEnum()
-            writer.insertTrailingNewline()
-            // impl From<str> for Blah { ... }
-            renderFromForStr()
-            // impl FromStr for Blah { ... }
-            renderFromStr()
-            writer.insertTrailingNewline()
-            // impl Blah { pub fn as_str(&self) -> &str
-            implBlock()
-            writer.rustBlock("impl AsRef<str> for $enumName") {
-                rustBlock("fn as_ref(&self) -> &str") {
-                    rust("self.as_str()")
-                }
-            }
-        } else {
-            renderUnnamedEnum()
-        }
-    }
-
-    private fun renderUnnamedEnum() {
-        writer.documentShape(shape, model)
-        writer.deprecatedShape(shape)
-        meta.render(writer)
-        writer.write("struct $enumName(String);")
-        writer.rustBlock("impl $enumName") {
-            docs("Returns the `&str` value of the enum member.")
-            rustBlock("pub fn as_str(&self) -> &str") {
-                rust("&self.0")
-            }
-
-            docs("Returns all the `&str` representations of the enum members.")
-            rustBlock("pub fn $Values() -> &'static [&'static str]") {
-                withBlock("&[", "]") {
-                    val memberList = sortedMembers.joinToString(", ") { it.value.dq() }
-                    rust(memberList)
-                }
-            }
-        }
-
-        writer.rustBlock("impl <T> #T<T> for $enumName where T: #T<str>", RuntimeType.From, RuntimeType.AsRef) {
-            rustBlock("fn from(s: T) -> Self") {
-                rust("$enumName(s.as_ref().to_owned())")
+        // TODO(https://github.com/awslabs/smithy-rs/issues/1700): handle IntEnum
```

Comment:
Does it generate invalid code or throw an exception when encountering an integer enum? If not, we should make it do so for now.

Reply:
Changed so that it throws an exception both here and when the conversion fails